zfsinstaller/zfsinstaller.cpp

```diff
@@ -47,22 +226,53 @@ int main(int argc, char* argv[])
 	if (strcmp(argv[1], "install") == 0) {
 		zfs_install(argv[2]);
 		fprintf(stderr, "Installation done.");
-	} else if(strcmp(argv[1], "uninstall") == 0) {
-		zfs_uninstall(argv[2]);
-	} else {
+	}
+	else if (strcmp(argv[1], "uninstall") == 0) {
+		int ret = zfs_uninstall(argv[2]);
+		if (0 == ret) {
+			return zfs_log_session_delete();
+		}
+		return ret;
+	}
+	else if (strcmp(argv[1], "trace") == 0) {
+		if (argc == 3 && strcmp(argv[2], "-d") == 0) {
+			int ret;
+			ret = zfs_log_session_delete();
+			if (ret == 0) {
+				fprintf(stderr, "Logman session %s deleted successfully\n", LOGGER_SESSION);
+			}
+			else {
+				fprintf(stderr, "Error while deleting session %s\n", LOGGER_SESSION);
+			}
+			return ret;
+		}
+		else {
+			return zfs_log_session_create(argc - 1, &argv[1]);
+		}
+	}
+	else {
 		fprintf(stderr, "unknown argument %s\n", argv[1]);
 		printUsage();
 		return ERROR_BAD_ARGUMENTS;
 	}
+	return 0;
 }
 
 void printUsage() {
 	fprintf(stderr, "Usage:\n\n");
 	fprintf(stderr, "Install driver per INF DefaultInstall section:\n");
-	fprintf(stderr, "zfsinstaller install [inf path]\n");
+	fprintf(stderr, "zfsinstaller install inf_path\n");
 	fprintf(stderr, "\n");
 	fprintf(stderr, "Uninstall driver per INF DefaultUninstall section:\n");
-	fprintf(stderr, "zfsinstaller uninstall [inf path]\n");
+	fprintf(stderr, "zfsinstaller uninstall inf_path\n");
+	fprintf(stderr, "\n");
+	fprintf(stderr, "zfsinstaller trace [-d] [-f Flags] | [-l Levels] | [-s SizeOfETLInMB] | [-p AbsolutePathOfETL]\n");

```

Comment:
Keep `-d` and other switches in separate lines, as the commands are different essentially.